src/Polly.SharedSpecs/Monkey/MonkeySpecs.cs

```diff
@@ -0,0 +1,539 @@
+
+            executed.Should().BeTrue();
+        }
+
+        [Fact]
+        public void Monkey_With_Context_Should_execute_user_delegate_2()
+        {
+            this.Init();
+            Exception fault = new Exception("test");
+            Boolean executed = false;
+            Context context = new Context();
+            context["ShouldFail"] = "false";
+            MonkeyPolicy policy = Policy.Monkey(
+                new Exception("test"),
+                0.6,
+                (ctx) =>
+                {
+                    return (ctx["ShouldFail"] != null && ctx["ShouldFail"].ToString() == "true");
+                });
+
+            policy.Invoking(x => x.Execute((ctx) => { executed = true; }, context))
+                .ShouldNotThrow<Exception>();
+
+            executed.Should().BeTrue();
+        }
+        #endregion
+
+        #region Overload, Exception based on context
+        [Fact]
+        public void Monkey_With_Context_Exception_Should_not_execute_user_delegate_1()
+        {
+            this.Init();
+            Boolean executed = false;
+            Context context = new Context();
+            context["ShouldFail"] = "true";
+
+            Func<Context, Exception> fault = (ctx) => new Exception("test");
+            Func<Context, bool> enabled = (ctx) => true;
+            MonkeyPolicy policy = Policy.Monkey(fault, 0.6, enabled);
+            policy.Invoking(x => x.Execute((ctx) => { executed = true; }, context))
+                .ShouldThrowExactly<Exception>();
+
+            executed.Should().BeFalse();
+        }
+
+        [Fact]
+        public void Monkey_With_Context_Exception_Should_not_execute_user_delegate_2()
+        {
+            this.Init();
+            Boolean executed = false;
+            Context context = new Context();
+            context["ShouldFail"] = "true";
+
+            Func<Context, Exception> fault = (ctx) =>
+            {
+                if (ctx["ShouldFail"] != null && ctx["ShouldFail"].ToString() == "true")
+                {
+                    return new InvalidOperationException();
+                }
+
+                return new InvalidProgramException("test");
+            };
+
+            Func<Context, bool> enabled = (ctx) => true;
+            MonkeyPolicy policy = Policy.Monkey(fault, 0.6, enabled);
+            policy.Invoking(x => x.Execute((ctx) => { executed = true; }, context))
+                .ShouldThrowExactly<InvalidOperationException>();
+
+            executed.Should().BeFalse();
+        }
+
+        [Fact]
+        public void Monkey_With_Context_Exception_Should_not_execute_user_delegate_3()
+        {
+            this.Init();
+            Boolean executed = false;
+            Context context = new Context();
+            context["ShouldFail"] = "false";
+
+            Func<Context, Exception> fault = (ctx) =>
+            {
+                if (ctx["ShouldFail"] != null && ctx["ShouldFail"].ToString() == "true")
```

Comment:
Variable here ^ is controlling which exception, not "ShouldFail" this time, maybe we should rename this one?

Reply:
Removed the overload in MonkeyPolicy and hence the test